indico/modules/events/management/controllers/cloning.py

```diff
@@ -101,21 +100,32 @@ class PatternCloneCalculator(CloneCalculator):
     def _calculate(self, form):
         args = {'dtstart': self._naivify(form.start_dt.data)}
         args.update(self._calc_stop_criteria(form))
-        return self._tzify(rrule.rrule(rrule.MONTHLY,
-                                       interval=form.num_months.data,
-                                       byweekday=form.week_day.week_day_data,
-                                       bysetpos=form.week_day.day_number_data,
-                                       **args))
+        return (self._tzify(rrule.rrule(rrule.MONTHLY,
+                                        interval=form.num_months.data,
+                                        byweekday=form.week_day.week_day_data,
+                                        bysetpos=form.week_day.day_number_data,
+                                        **args)),
+                False)
 
 
 class IntervalCloneCalculator(CloneCalculator):
     form_class = CloneRepeatIntervalForm
 
     def _calculate(self, form):
-        args = {'dtstart': self._naivify(form.start_dt.data)}
-        args.update(self._calc_stop_criteria(form))
+        last_day_of_month = False
         freq, interval = relativedelta_to_rrule_interval(form.recurrence.data)
-        return self._tzify(rrule.rrule(freq, interval=interval, **args))
+        # check if last day of month
+        dtstart = self._naivify(form.start_dt.data)
+        next_day = dtstart + timedelta(days=1)
+        if freq == rrule.MONTHLY and next_day.day == 1:
+            args = {'dtstart': next_day}
+            args.update(self._calc_stop_criteria(form))
+            dates = rrule.rrule(freq, interval=interval, **args)
+            last_day_of_month = True
```

Comment:
why not just add `True` in the next line (and `False` further below)?

Reply:
or, you could assign `dates = self._tzify(...)`, move the stuff after the return into an `else` block, and below all that `return dates, last_day_of_month`

Reply:
I can assign the dates, I didn't want to pass simply True/False to be more verbose about the parameter

Reply:
I think this kind of verbosity would rather belong into a docstring, e,g, like this

    :return: a ``(dates, last_day_of_month)`` tuple